mcs/class/test-helpers/SocketResponder.cs

```diff
@@ -91,8 +91,11 @@ private void Listen ()
 					listenSocket = tcpListener.AcceptSocket ();
 					listenSocket.Send (requestHandler (listenSocket));
 					try {
-						listenSocket.Shutdown (SocketShutdown.Receive);
+						// On Windows a Receive() is needed here before Shutdown() to consume the data some tests send.
```

Comment:
is this really a "On Windows" issue or a general race that just happens to almost never occur on our Unix implementation? If the latter, I'd remove the windows bit from the comment.